### PR TITLE
stream_info: add setResponseCode and update local_reply to take a normal StreamInfo

### PR DESCRIPTION
**Pull request by esmet**
Commit Message: stream_info: add setResponseCode and update local_reply to take a normal StreamInfo
Additional Description: This makes local_reply a little cleaner by removing a small implementation detail and adding a new setter function to StreamInfo.
Risk Level: medium (refactors existing code)
Testing: No new tests
Docs Changes: None
Release Notes: None

### CODE REVIEW

File: include/envoy/stream_info/stream_info.h [hunk 1]
```diff
@@ -236,6 +236,11 @@ class StreamInfo {
    */
   virtual void setResponseFlag(ResponseFlag response_flag) PURE;
 
+  /**
+   * @param code the HTTP response code to set for this request.
+   */
+  virtual void setResponseCode(uint32_t code) PURE;
```
Comment (esmet) [inline include/envoy/stream_info/stream_info.h:242]: The callers of this function pass in Http::Code but the underlying type is uint32_t, so I went with that.

Open to suggestions as to whether we think this should remain uint32_t or change to Http::Code.

**Comment by alyssawilk** [comment]
LG overall - want to take a look at CI and ping back?

**Comment by esmet** [comment]
> LG overall - want to take a look at CI and ping back?

Yes it looks like I missed updating a test. Will do shortly and ping back.

**Review by alyssawilk — APPROVED** [review]

**Comment by esmet** [comment]
Pushed the fix and CI looks 💚

**Review by alyssawilk — APPROVED** [review]
LGTM!  @snowp want to take a look as well?

**Review by snowp — APPROVED** [review]
LGTM, thanks!

**Pull request merged**